### ISSUE TITLE
JIT: Bad codegen with `Bmi1.BitFieldExtract`

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#105820]
```csharp
// Generated by Fuzzlyn v2.2 on 2024-08-01 15:19:29
// Run on X64 Linux
// Seed: 10778112854935688556-vectort,vector128,vector256,x86aes,x86avx,x86avx2,x86avx512bw,x86avx512bwvl,x86avx512cd,x86avx512cdvl,x86avx512dq,x86avx512dqvl,x86avx512f,x86avx512fvl,x86avx512fx64,x86bmi1,x86bmi1x64,x86bmi2,x86bmi2x64,x86fma,x86lzcnt,x86lzcntx64,x86pclmulqdq,x86popcnt,x86popcntx64,x86sse,x86ssex64,x86sse2,x86sse2x64,x86sse3,x86sse41,x86sse41x64,x86sse42,x86sse42x64,x86ssse3,x86x86base
// Reduced from 23.7 KiB to 0.5 KiB in 00:00:37
// Debug: Prints 0 line(s)
// Release: Prints 1 line(s)
using System;
using System.Runtime.CompilerServices;
using System.Numerics;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.X86;

public class Program
{
    public static void Main()
    {
        var vr12 = (byte)0;
        if ((int)Bmi1.BitFieldExtract(3774778286U, vr12, 255) >= 0)
        {
            var vr13 = Vector128.CreateScalar(1);
            Vector128<long> vr14 = Sse41.ConvertToVector128Int64(vr13);
            System.Console.WriteLine(vr14);
        }
    }
}
```

cc @dotnet/jit-contrib

**Comment by JulieLeeMSFT** [comment]
@anthonycanino, PTAL. CC @tannergooding.

**Comment by khushal1996** [comment]
Brief analysis of this issue -->

```
public static void Issue105820()
{
    var vr12 = (byte)0;
    if ((int)Bmi1.BitFieldExtract(3774778286U, vr12, 255) >= 0)
    {
        System.Console.WriteLine("Wrong Wrong Wrong Wrong ********************");
    }
}
```

**Debug Output** -->
```
```

**Release output**  -->
```
Wrong Wrong Wrong Wrong ********************
```

Debug Disasm -->
```
; Assembly listing for method Program:Issue105820() (Tier0)
; Emitting BLENDED_CODE for X64 with AVX512 - Windows
; Tier0 code
; rbp based frame
; partially interruptible

G_M000_IG01:                ;; offset=0x0000
       55                   push     rbp
       4883EC30             sub      rsp, 48
       488D6C2430           lea      rbp, [rsp+0x30]
       33C0                 xor      eax, eax
       8945FC               mov      dword ptr [rbp-0x04], eax
 
G_M000_IG02:                ;; offset=0x000F
       33C0                 xor      eax, eax
       8945FC               mov      dword ptr [rbp-0x04], eax
       8B55FC               mov      edx, dword ptr [rbp-0x04]
       B9AE8BFEE0           mov      ecx, 0xFFFFFFFFE0FE8BAE
       41B8FF000000         mov      r8d, 255
       FF15202BF7FF         call     [System.Runtime.Intrinsics.X86.Bmi1:BitFieldExtract(uint,ubyte,ubyte):uint]
       85C0                 test     eax, eax
       7C10                 jl       SHORT G_M000_IG03
       48B980245BCECC010000 mov      rcx, 0x1CCCE5B2480
       FF153CBD1700         call     [System.Console:WriteLine(System.String)]
 
G_M000_IG03:                ;; offset=0x003C
       90                   nop      
 
G_M000_IG04:                ;; offset=0x003D
       4883C430             add      rsp, 48
       5D                   pop      rbp
       C3                   ret      
 
; Total bytes of code 67
```

**Release disasm** -->
```
; Assembly listing for method Program:Issue105820() (FullOpts)
; Emitting BLENDED_CODE for X64 with AVX512 - Windows
; FullOpts code
; optimized code
; rsp based frame
; partially interruptible
; No PGO data
; 0 inlinees with PGO data; 1 single block inlinees; 0 inlinees without PGO data

G_M000_IG01:                ;; offset=0x0000
       4883EC28             sub      rsp, 40
 
G_M000_IG02:                ;; offset=0x0004
       B900FF0000           mov      ecx, 0xFF00
       B8AE8BFEE0           mov      eax, 0xFFFFFFFFE0FE8BAE
       C4E270F7C8           bextr    ecx, eax, ecx
       7C10                 jl       SHORT G_M000_IG04
 
G_M000_IG03:                ;; offset=0x0015
       48B9002013C066020000 mov      rcx, 0x266C0132000
       FF159B801B00         call     [System.Console:WriteLine(System.String)]
 
G_M000_IG04:                ;; offset=0x0025
       90                   nop      
 
G_M000_IG05:                ;; offset=0x0026
       4883C428             add      rsp, 40
       C3                   ret      
 
; Total bytes of code 43
```

Probable cause --> The ```test eax, eax``` instruction before the ```jl``` instruction is folded somewhere in release mode which prevents the setting of SF (Sign flag) and hence the release code goes inside the if loop. ```bextr``` does not set those flags and hence I am looking at the flag settings and how this code is optimized in release mode.

**Comment by amanasifkhalid** [comment]
Fixed by #106440, backported to release/9.0
